File: leetcode-java-scala-solutions/src/main/scala/br/com/mystudies/leetcode/scala/solutions/easy/ReverseInteger.scala
package br.com.mystudies.leetcode.scala.solutions.easy

class ReverseInteger {

  def reverse(x: Int) = {
    try {
    	if(x < 0 ){
    		(-x).toString.reverse.toInt.unary_-
    	}else{
    		x.toString.reverse.toInt
    	}
    } catch {
      case t: NumberFormatException => 0
    }
  }


}

File: leetcode-java-scala-solutions/src/main/java/br/com/mystudies/leetcode/java/solutions/easy/ReverseInteger.java
package br.com.mystudies.leetcode.java.solutions.easy;

public class ReverseInteger {
	public int reverse(int x) {
		try {
			if(x < 0){
				return -Integer.valueOf(toStringReverse(-x));
			}
			return Integer.valueOf(toStringReverse(x));
		} catch (NumberFormatException e) {
			return 0;
		}
	}
	private String toStringReverse(int x) {
		return new StringBuilder(Integer.toString(x)).reverse().toString();
	}
}
